crates/core/transaction/src/transaction.rs

```diff
@@ -445,6 +469,66 @@ impl From<TransactionBody> for Vec<u8> {
     }
 }
 
+impl TypeUrl for TransactionParameters {
+    const TYPE_URL: &'static str = "/penumbra.core.transaction.v1alpha1.TransactionParameters";
+}
+
+impl DomainType for TransactionParameters {
+    type Proto = pbt::TransactionParameters;
+}
+
+impl TryFrom<pbt::TransactionParameters> for TransactionParameters {
+    type Error = Error;
+
+    fn try_from(proto: pbt::TransactionParameters) -> anyhow::Result<Self, Self::Error> {
+        Ok(TransactionParameters {
+            expiry_height: proto.expiry_height,
+            chain_id: proto.chain_id,
+        })
+    }
+}
+
+impl From<TransactionParameters> for pbt::TransactionParameters {
+    fn from(msg: TransactionParameters) -> Self {
+        pbt::TransactionParameters {
+            expiry_height: msg.expiry_height,
+            chain_id: msg.chain_id,
+        }
+    }
+}
+
+impl TypeUrl for DetectionData {
+    const TYPE_URL: &'static str = "/penumbra.core.transaction.v1alpha1.DetectionData";
+}
+
+impl DomainType for DetectionData {
+    type Proto = pbt::DetectionData;
+}
+
+impl TryFrom<pbt::DetectionData> for DetectionData {
+    type Error = Error;
+
+    fn try_from(proto: pbt::DetectionData) -> anyhow::Result<Self, Self::Error> {
+        let fmd_clues = proto
+            .fmd_clues
+            .into_iter()
+            .map(|x| x.try_into())
+            .collect::<Result<Vec<Clue>, Error>>()?;
+        Ok(DetectionData { fmd_clues })
+    }
+}
+
+impl From<DetectionData> for pbt::DetectionData {
+    fn from(msg: DetectionData) -> Self {
+        let mut fmd_clues = Vec::new();
+        for fmd_clue in msg.fmd_clues {
+            fmd_clues.push(fmd_clue.into());
+        }
+
+        pbt::DetectionData { fmd_clues }
```

Comment:
Maybe we could have this match the `TryFrom` impl above, using an iterator? Not sure what convention is preferred.

```suggestion
        let fmd_clues = proto
            .fmd_clues
            .into_iter()
            .map(|x| x.try_into())
            .collect();
            
        pbt::DetectionData { fmd_clues }
```